source/common/upstream/upstream_impl.h

```diff
@@ -142,6 +146,7 @@ class HostDescriptionImpl : virtual public HostDescription {
   HostStats stats_;
   Outlier::DetectorHostMonitorPtr outlier_detector_;
   HealthCheckHostMonitorPtr health_checker_;
+  uint32_t priority_;
```

Comment:
I think this needs to be atomic much like weight since you modify it inline?

Reply:
Ah yeah that would make sense. I assume this means that the host objects are shared between threads?

Reply:
Yup